src/wolff_producer.erl

```diff
@@ -460,43 +511,46 @@ handle_kafka_ack(#kpro_rsp{api = produce,
   BaseOffset = kpro:find(base_offset, PartitionRsp),
   case queue:peek(SentReqs) of
       {value, #{request := #kpro_req{ref = Ref},
-                batch_size := BatchSize,
                 attempts := Attempts}} ->
           case ErrorCode =:= ?no_error of
               true ->
                   do_handle_kafka_ack(BaseOffset, St);
               false ->
                   AttemptedBefore = Attempts > 1,
-                  inc_sent_failed(Config, BatchSize, AttemptedBefore),
-                  wolff_metrics:inflight_change(Config, -BatchSize),
-                  #{topic := Topic, partition := Partition} = St,
+                  #{ topic := Topic
+                   , sent_reqs_count := SentReqsCount
+                   , partition := Partition
+                   } = St,
+                  inc_sent_failed(Config, 1, AttemptedBefore),
+                  wolff_metrics:inflight_set(Config, SentReqsCount - 1),
                   log_warn(Topic, Partition,
                            "error_in_produce_response",
                            #{error_code => ErrorCode}),
                   erlang:throw(ErrorCode)
           end;
       _ ->
-          ok 
+          St
```

Comment:
oops.

Reply:
which versions did this impact? might be good to keep a record in changelog

Reply:
this is usually not an issue, unless there is a bug?

Reply:
> this is usually not an issue, unless there is a bug?

probably.  unless a `{msg, _, _}` message is sent to the producer while its queue is empty for whatever reason.

Reply:
added an entry to the changelog